cou/apps/base.py

```diff
@@ -912,29 +912,35 @@ def _check_application_target(self, target: OpenStackRelease) -> None:
             )
 
     def _check_mismatched_versions(self) -> None:
-        """Check that there are no mismatched versions on app units.
+        """Check that there are no unexpected mismatched versions on app units.
+
+        For cases where mismatched versions may be expected
+        (eg. nova-compute which is upgraded unit-by-unit),
+        then it will not check.
 
         :raises MismatchedOpenStackVersions: When the units of the app are running
                                              different OpenStack versions.
         """
-        # NOTE (gabrielcocenza) nova-compute is upgraded using paused-single-unit,
-        # so it's possible to have mismatched version in applications units that are
-        # nova-compute or colocated with it.
+        # nova-compute is upgraded one unit at a time,
+        # so it's possible to have mismatched version in
+        # units of applications that are nova-compute or colocated with it.
         if any(
             "nova-compute" in app_charm
             for machine in self.machines.values()
             for app_charm in machine.apps_charms
         ):
             return
 
-        o7k_versions = self.o7k_release_units
-        if len(o7k_versions.keys()) > 1:
-            mismatched_repr = [
-                f"'{openstack_release.codename}': {units}"
-                for openstack_release, units in o7k_versions.items()
-            ]
-
+        if len({self.get_latest_o7k_version(unit) for unit in self.units.values()}) > 1:
```

Comment:
I decided against using `self.o7k_release_units` in this method, because the data structure it returns is awkward in the context and it doesn't provide the actual workload versions.

I could use it just for this conditional, but I'd rather be consistent with the following code, than use two separate methods.

Reply:
Solving as this is not a blocker